azurelinuxagent/ga/update.py

```diff
@@ -1173,16 +1172,3 @@ def _execute_run_command(command):
         except Exception as e:
             msg = "Error while checking ip table rules:{0}".format(ustr(e))
             logger.error(msg)
-
-    def _reset_legacy_blacklisted_agents(self):
```

Comment:
This reset putting agent update in loop (because it is removing error file) and also, I don't see a value now(how it helps). we still need info of blacklisted agents that are blacklisted by daemon.

Reply:
at some point the blacklisting logic in the daemon changed and seems like this code was trying to get rid of errors from previous daemons

with your changes we are removing all agents other than (current, update), whether they are blacklisted or not, right?

if yes, i agree that it is safe to remove this

Reply:
yup, we remove everything else no matter what state they are